packages/patch/src/errors.ts

```diff
@@ -0,0 +1,185 @@
+import { Kind } from 'graphql';
+import type { Change } from '@graphql-inspector/core';
+import type { PatchConfig } from './types.js';
+
+export function handleError(change: Change<any>, err: Error, config: PatchConfig) {
+  if (config.onError?.(err, change) === true) {
+    // handled by onError
+    return;
+  }
+
+  if (err instanceof NoopError) {
+    console.debug(
+      `Ignoring change ${change.type} at "${change.path}" because it does not modify the resulting schema.`,
+    );
+  } else if (!config.requireOldValueMatch && err instanceof ValueMismatchError) {
+    console.debug(`Ignoring old value mismatch at "${change.path}".`);
+  } else if (config.throwOnError === true) {
+    throw err;
+  } else {
+    console.warn(`Cannot apply ${change.type} at "${change.path}". ${err.message}`);
+  }
+}
+
+/**
+ * When the change does not actually modify the resulting schema, then it is
+ * considered a "no-op". This error can safely be ignored.
+ */
+export class NoopError extends Error {
+  readonly noop = true;
+  constructor(message: string) {
+    super(`The change resulted in a no op. ${message}`);
+  }
+}
+
+export class ValueMismatchError extends Error {
+  readonly mismatch = true;
+  constructor(kind: Kind, expected: string | undefined | null, actual: string | undefined | null) {
+    super(
+      `The existing value did not match what was expected. Expected the "${kind}" to be ${expected} but found ${actual}.`,
+    );
+  }
+}
+
+/**
+ * If the requested change would not modify the schema because that change is effectively
+ * already applied.
+ *
+ * If the added coordinate exists but the kind does not match what's expected, then use
+ * ChangedCoordinateKindMismatchError instead.
+ */
+export class AddedCoordinateAlreadyExistsError extends NoopError {
+  constructor(
+    public readonly kind: Kind,
+    readonly expectedNameOrValue: string | undefined,
+  ) {
+    const expected = expectedNameOrValue ? `${expectedNameOrValue} ` : '';
+    super(`A "${kind}" ${expected}already exists at the schema coordinate.`);
+  }
+}
+
+export type NodeAttribute =
+  | 'description'
+  | 'defaultValue'
+  /** Enum values */
+  | 'values'
+  /** Union types */
+  | 'types'
+  /** Return type */
+  | 'type'
```

Comment:
Could this be more explicit?
```suggestion
  /** Return type */
  | 'returnType'
```